lapis2/src/test/kotlin/org/genspectrum/lapis/controller/MockData.kt

```diff
@@ -0,0 +1,308 @@
+            AGGREGATED_ROUTE -> aggregated
+            NUCLEOTIDE_MUTATIONS_ROUTE -> nucleotideMutations
+            AMINO_ACID_MUTATIONS_ROUTE -> aminoAcidMutations
+            NUCLEOTIDE_INSERTIONS_ROUTE -> nucleotideInsertions
+            AMINO_ACID_INSERTIONS_ROUTE -> aminoAcidInsertions
+            else -> throw IllegalArgumentException("Test issue: no mock data for endpoint $endpoint")
+        }
+
+    val fastaMockData = MockData.createForFastaEndpoint(
+        """
+            >sequence1
+            CAGAA
+            >sequence2
+            CAGAA
+        """.trimIndent(),
+    )
+
+    private val aggregated = MockDataCollection.create(
+        siloQueryModelMockCall = { it::getAggregated },
+        modelData = listOf(
+            AggregationData(
+                0,
+                mapOf("country" to TextNode("Switzerland"), "age" to IntNode(42)),
+            ),
+        ),
+        expectedJson = """
+            [
+                {
+                    "country": "Switzerland",
+                    "age": 42,
+                    "count": 0
+                }
+            ]
+        """.trimIndent(),
+        expectedCsv = """
+            country,age,count
+            Switzerland,42,0
+        """.trimIndent(),
+        expectedTsv = """
+           country	age	count
+           Switzerland	42	0
+        """.trimIndent(),
+    )
+
+    private val details = MockDataCollection.create(
+        siloQueryModelMockCall = { it::getDetails },
+        modelData = listOf(
+            DetailsData(
+                mapOf(
+                    "country" to TextNode("Switzerland"),
+                    "age" to IntNode(42),
+                    "floatValue" to DoubleNode(3.14),
+                ),
+            ),
+            DetailsData(
+                mapOf(
+                    "country" to TextNode("Switzerland"),
+                    "age" to IntNode(43),
+                    "floatValue" to NullNode.instance,
+                ),
+            ),
+        ),
+        expectedJson = """
+            [
+                {
+                    "country": "Switzerland",
+                    "age": 42,
+                    "floatValue": 3.14
+                },
+                {
+                    "country": "Switzerland",
+                    "age": 43,
+                    "floatValue": null
+                }
+            ]
+        """.trimIndent(),
+        expectedCsv = """
+            country,age,floatValue
+            Switzerland,42,3.14
+            Switzerland,43,null
+        """.trimIndent(),
+        expectedTsv = """
+            country	age	floatValue
+            Switzerland	42	3.14
+            Switzerland	43	null
+        """.trimIndent(),
+    )
+
+    private val nucleotideMutations = MockDataCollection.create(
+        siloQueryModelMockCall = { it::computeNucleotideMutationProportions },
+        modelData = listOf(
+            NucleotideMutationResponse(
+                "sequenceName:1234",
```

Comment:
Do we want to add the symbolFrom and symbolTo, since this is used in many tests? Or is this the more difficult version to handle?

Reply:
Actually it doesn't matter, because we don't parse the response, but I'll add it for clarity.